### ISSUE TITLE
Open multiple requests

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#129]
* As an intake analyst
* I want to open multiple requests at once
* so that I can save time

**Assumptions & Scope**	
What are the assumptions for this story?

Ticket needs to have been saved (all mandatory fields including assigned to need to be complete)
AXIS integration is not in scope.

What is in and not in scope?
The creation of the Acknowledgement Letter with the multiple file numbers is not in scope.
Printing not in scope #127
Days remaining (green button, top left of the correspondence log response view) is out of scope

**Acceptance Criteria**	

**Scenario 1: Open multiple ministry requests**
* GIVEN the analyst is on the Split Request view
* https://jacklyn808742.invisionapp.com/console/FOI---Intake-ckpq19frj032r013q6gpm8t1m/ckpx6mtxp012f013ebi30csot/play
* AND more than one ministry has been selected
* AND the request descriptions have been filled
* AND the date ranges have been specified
* WHEN the analyst clicks in the 'Open' button
* THEN the Open Multiple Ministry Requests modal will open
* https://jacklyn808742.invisionapp.com/console/FOI---Intake-ckpq19frj032r013q6gpm8t1m/ckpx6mtxp012h013egppn0ra5/play
* AND a file number will have been generated for each ministry selected

**Scenario 2: Cllck 'Create Acknowledgement Letter'**
* GIVEN an intake analyst is on the Open Multiple Ministry Requests modal will open
* https://jacklyn808742.invisionapp.com/console/FOI---Intake-ckpq19frj032r013q6gpm8t1m/ckpx6mtxp012h013egppn0ra5/play
*AND the applicant has provided an email address
* WHEN the applicant clicks the 'Create Acknowledgement Letter' button
* THEN the Correspondence View will open
* https://jacklyn808742.invisionapp.com/console/FOI---Intake-ckpq19frj032r013q6gpm8t1m/ckpybvbw802h7014wf77hd2he/play
* AND multiple request numbers will be the email subject line
* AND the editable email template will be pre-filled
* AND the acknowledgement letter will be attached
* AND the applicant's email will be pre-filled

**Scenario 3: Send Message and Open Requests'**
* GIVEN an intake analyst is Acknowledgement Correspondence View will open
* https://jacklyn808742.invisionapp.com/console/FOI---Intake-ckpq19frj032r013q6gpm8t1m/ckpybvbw802h7014wf77hd2he/play
* WHEN the analyst clicks 'Send Message'
* THEN the email will be sent to the applicant with the attached Acknowledgement Letter
* AND a toast will briefly appear that the email has been sent
* AND multiple requests will be opened, each with a file number
* AND the analyst will be returned to the Request Queue
* AND the requests will be in the 'Opened' status
* https://jacklyn808742.invisionapp.com/console/FOI---Intake-ckpq19frj032r013q6gpm8t1m/ckpq19gnk00p0011q7zsk81wb/play

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**
Ticket #104 (contact applicant) is a prerequisite. UI from that ticket will be used here.

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by JHarrietha-AOT** [comment]
@lmullane @mpilchar This issue looks good to me

**Comment by lmullane** [comment]
Thanks, @JHarrietha-AOT

**Comment by abin-aot** [comment]
1. " file number will have been generated for each ministry selected" , Are we planning to cover AXIS integration here? Because FILE NUMBER should have a match with AXIS, since we are planning to run this parallelly for a while - Right?

2. What will happen to "Create Acknowledge letter" modal pop up if there is no "Email" ? Are covering this on this User story. Because to open a request we need "email" as per scenario 3 

3.  Suggestion : Seems like Scenario 2 and 3 , can be separate user stories, considering the point that there are much effort behind to create PDF attachments only the fly from a rich text editor, preview etc.

4. Diff. between scenario 1 and 4 ?

**Comment by lmullane** [comment]
@abin-aot 

1. AXIS integration is not in scope - I don't expect we will be able to integrate for months.
2. @JHarrietha-AOT, there should be an Exception with a popup stating this action cannot be completed without and email. They will need to get an email or download the letter
3. The creation of all templates, including email and letter templates , are not in scope for this story. This is only the user interface.
4. Yes, they should be mandatory. Assignment will be a separate story that is not in scope for this story. You cannot Open a Request until your Save the Category and assign someone to the request.
5. 4 shouldn't have been there. Axed it.
6. Workflow would change from Intake to Open
7. Yes

**Comment by mpilchar** [comment]
@abin-aot and @lmullane thanks for these Q&As! Helpful questions, and helpful answers. Looking forward to chatting more tomorrow

**Comment by mpilchar** [comment]
Dependent on business decision / reliance on AXIS

**Comment by mpilchar** [comment]
completed - closing
